### PR TITLE
CDAP-2012: Add application version in the ApplicationSpec

### PR DESCRIPTION
**Pull request by jwang47**
http://builds.cask.co/browse/CDAP-DUT1354

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ApplicationSpecificationCodec.java [hunk 1]
```diff
@@ -70,6 +71,7 @@ public ApplicationSpecification deserialize(JsonElement json, Type typeOfT, Json
     JsonObject jsonObj = json.getAsJsonObject();
 
     String name = jsonObj.get("name").getAsString();
+    String version = jsonObj.get("version").getAsString();
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ApplicationSpecificationCodec.java]: What if it is missing? E.g. existing spec in MDS?

**Comment by sreevatsanraman** [comment]
Add unit test.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/DefaultApplicationSpecification.java [hunk 2]
```diff
@@ -93,6 +95,11 @@ public String getName() {
   }
 
   @Override
+  public String getVersion() {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/DefaultApplicationSpecification.java]: Seems like version is `nullable`? What happen if the jar manifest doesn't have `Bundle-Version`?

**Comment by awholegunch** [comment]
Does this change the examples? For instance, in the example Application of the Developer's Manual: Building Block, would we add "setVersion"?

**Comment by jwang47** [comment]
No, this won't allow developers to set the version of their app. It will the pull the version from the jar that they deploy (version is set from jar manifest).

**Comment by sreevatsanraman** [comment]
LGTM :+1:

**Pull request merged**